control-plane/p4RuntimeArchHandler.h

```diff
@@ -403,8 +403,9 @@ struct Counterlike {
             auto typeArg = type->arguments->at(*indexTypeParamIdx);
             // We ignore the return type on purpose, but the call is required to update p4RtTypeInfo
             // if the index has a user-defined type.
-            TypeSpecConverter::convert(refMap, typeMap, typeArg, p4RtTypeInfo);
-            index_type_name = getTypeName(typeArg, typeMap);
+            if (!dynamic_cast<const IR::Type_Dontcare *>(typeArg)) {
+                TypeSpecConverter::convert(refMap, typeMap, typeArg, p4RtTypeInfo);
```

Comment:
a better solution may be to handle `Type_Dontcare` in the `TypeSpecConverter` visitor?

also is there any PSA / v1model program that can illustrate this case? I'm having trouble figuring out what `Type_Dontcare` means here. If this code is exercised by your internal compiler, did you add the correct `indexTypeParamIdx` methods (see https://github.com/p4lang/p4c/commit/d7783e1036de5c1d2e6b65cdc2d1c3bc47db68ed#diff-36306e8998b70ba36e5831d493ce90f4)?

Reply:
I modified the psa-basic-counter-bmv2.p4 to show this.  Basically, requiring the user to come up with a bit size for the indexes is painful -- they generally just want a size that is 'big enough' and efficient on the target (which may vary from target to target).  Dontcare (`_`) seeme like a reasonable way of saying this, and had been working previously.

Reply:
Interesting, I wasn't aware of this feature. Then it seems that you should be able to add a PSA test case as part of your PR so we don't break this in the future.